src/xcoobee/api/PolicyApi.js

```diff
@@ -15,26 +15,20 @@ import UploadPolicyIntents from './UploadPolicyIntents';
  *
  * @throws {XcooBeeError}
  */
-export function upload_policy(apiUrlRoot, apiAccessToken, intent, endPointCursor, files) {
+const upload_policy = (apiUrlRoot, apiAccessToken, intent, endPointCursor, files) => {
   if (!UploadPolicyIntents.values.includes(intent)) {
-    throw TypeError('`intent`' + ` must be one of ${UploadPolicyIntents.values.join(', ')}.`);
+    return Promise.reject(new TypeError(`'intent' must be one of ${UploadPolicyIntents.values.join(', ')}.`));
   }
-  if (!appearsToBeACursor(endPointCursor)) {
-    throw TypeError('`endPointCursor` is required.');
+  if (!ApiUtils.appearsToBeACursor(endPointCursor)) {
+    return Promise.reject(new TypeError('`endPointCursor` is required.'));
   }
   if (!Array.isArray(files)) {
-    throw TypeError('`files` must be an array.');
+    return Promise.reject(new TypeError('`files` must be an array.'));
   }
 
   let query = ['query uploadPolicy {'];
   files.forEach((file, idx) => {
-    let baseName;
-    if (file instanceof File) {
-      baseName = file.name;
-    }
-    else {
-      baseName = Path.basename(file);
-    }
+    const baseName = Path.basename(file);

```

Comment:
I think we need to add back in the check for `file instanceof` File`. This was added to make this work in a browser.